Task_22/Program.cs

```diff
@@ -0,0 +1,95 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task_22
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.OutputEncoding = Encoding.UTF8;
+
+            //Susikurkite savaitės temperatūrų rodmenų masyvą.Raskite:
+            //žemiausią temperatūrą;
+            //didžiausią temperatūrą;
+            //vidurkį;
+            //Temperatūrų, esančių žemiau už vidurkį, kiekį;
+            //Temperatūrų, esančių aukščiau už vidurkį, kiekį.
+
+            var temperatura = new int[] {9,10,10,8,4,4,5};
+
+            var min = temperatura.Min();
+            var max = temperatura.Max();
+            var suma = temperatura.Sum();
+            var vidurkis = suma / temperatura.Length;
+            var minSuma = 0;
+            var minKiekis = 0;
+            var maxSuma = 0;
+            var maxKiekis = 0;
+
+            for (var i = 0; i < temperatura.Length; i++)
+            {
+                if (temperatura[i] < vidurkis)
+                {
+                    minKiekis++;
+                    minSuma += temperatura[i];
+                }
+            }
+
+            for (var i = 0; i < temperatura.Length; i++) 
+            { 
+                if (temperatura[i] > vidurkis)
+                {
+                    maxKiekis++;
+                    maxSuma += temperatura[i];
+                }
+            }
+
+            Console.WriteLine("1.");
+            Console.WriteLine($"Savaitės temperaturos rodmenys: {temperatura[0]}, {temperatura[1]}, {temperatura[2]}, {temperatura[3]}, {temperatura[4]}, {temperatura[5]}, {temperatura[6]}");
+            Console.WriteLine($"žemiausia temperatūra: {min}");
+            Console.WriteLine($"didžiausia temperatūra: {max}");
+            Console.WriteLine($"Vidurkis yra: {vidurkis}");
+            Console.WriteLine($"Temperatūrų, esančių žemiau už vidurkį, kiekis yra: {minSuma} / {minKiekis} = {minSuma / minKiekis}");
+            Console.WriteLine($"Temperatūrų, esančių daugiau už vidurkį, kiekis yra: {maxSuma} / {maxKiekis} = {maxSuma / maxKiekis}");
+            Console.WriteLine();
+
+            //Susikurkite studentų pažymių masyvą. Raskite:
+            //geriausią pažymį;
+            //kiek mokinių yra gavę dešimtukus;
+            //kiek mokinių yra gavę neigiamą pažymį(žemiau nei 4);
+            //pažymių vidurkį.
+
+            var pazymys = new int[] { 10,9,8,10,7 };
+            var maxPazymys = pazymys.Max();
+            var maxPazymysKiekis = 0;
+            var minPazymysKiekis = 0;
+
+            for (var i = 0; i < pazymys.Length; i++)
+            {
+                if (pazymys[i] == 10)
+                {
+                    maxPazymysKiekis++;
+                }
+            }
+
+            for (var i = 0; i < pazymys.Length; i++)
+            {
+                if (pazymys[i] < 4)
+                {
+                    minPazymysKiekis++;
+                }
```

Comment:
Šią if sąlygą būtų galima įdėti kartu su dešimtukų kiekio tikrinimu, tuomet ciklas per visus pažymius vyktų vieną kartą. Įsivaizduokite, kad turite labai daug reikšmių kurias reikia patikrinti. Daryti ciklą kuris žiūri į visas reikšmes iš eilės idealiu atveju norime vieną kartą ir to pačio ciklo metu galime atlikti keletą skirtingų patikrinimų.